### PR TITLE
[CI] Mandatory windows support for all the versions

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Mandatory stages in the CI for older windows versions on a PR basis, similar to what it runs on `master/7.x/7.[0-9]+`

## Why is it important?

Discover incompatibility issues earlier rather than when merged. 

## Further details

`Jenkinsfile.yml` supports stages with multiple platforms, if we use that approach then the test output format will change and might affect the Test Dashboards regarding the top-10 failures or the flakiness analysis. 

If we do so, we might need to run some migration.

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

**Comment by v1v** [comment]
Linting caused by another PR that did not change the x-pack, and test failure unrelated to this changes.

File: .ci/scripts/generate_build_table.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
                     if "withModule" in doc["stages"][stage]:
                         withModule = doc["stages"][stage]["withModule"]
                     if "when" in doc["stages"][stage]:
-                        when = f"optional"
+                        if "not_changeset_full_match" not in doc["stages"][stage]["when"]:
+                            when = f"optional"
```
Comment (cachedout) [inline .ci/scripts/generate_build_table.py:35]:
```suggestion
                            when = "optional"
```
Nit: strictly speaking, it's a PEP8 violation to use a f-string if there is no interpolation.

**Review by cachedout — APPROVED** [review]

**Pull request merged**